iron-fit-behavior.html

```diff
@@ -414,8 +430,15 @@
         Math.min(top, bottom - this._fitInfo.sizedBy.minHeight));
 
       // Use right/bottom to set maxWidth/maxHeight, and respect minWidth/minHeight.
-      this.sizingTarget.style.maxWidth = Math.max(right - left, this._fitInfo.sizedBy.minWidth) + 'px';
-      this.sizingTarget.style.maxHeight = Math.max(bottom - top, this._fitInfo.sizedBy.minHeight) + 'px';
+      var maxWidth = Math.max(right - left, this._fitInfo.sizedBy.minWidth);
+      var maxHeight = Math.max(bottom - top, this._fitInfo.sizedBy.minHeight);
+
+      // Determine if scrollbar will display and increase max size if necessary
+      if (maxHeight < this.sizingTarget.clientHeight) { maxWidth += this._scrollbarSize; }
```

Comment:
Calling `clientHeight/clientWidth` would force layout on `sizingTarget`. You can store these values in `this._fitInfo` object, which gets updated only once and reused. The place where we measure the sizingTarget is in the `_discoverInfo` method. Over there you could add a new key to _fitInfo, e.g. `this._fitInfo.sizerDimensions = { clientWidth, clientHeight}` and read them here